### PR TITLE
[LM-10552] added UI layer in core

### PR DESCRIPTION
**Pull request by prabha-likeminds**
## Description

**Features**
1. Removed prop drilling in UI components

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fixes a breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: example/SampleApp/sampleApp/ios/Podfile.lock [hunk 1]
```diff
@@ -944,6 +944,21 @@ PODS:
   - React-Mapbuffer (0.73.1):
     - glog
     - React-debug
```
Comment (jaixchaudhary) [inline example/SampleApp/sampleApp/ios/Podfile.lock:946]: remove this file

File: likeminds-feed-reactnative-integration/components/LMMedia/LMCarousel/index.tsx [hunk 2]
```diff
@@ -66,7 +67,7 @@ const LMCarousel = React.memo(({
       renderItem={({ item, index }) => (
         <>
           {/* this section render image */}
-          {item?.attachmentType === ATTACHMENT_TYPE.IMAGE && (
+          {item?.attachmentType === IMAGE_ATTACHMENT_TYPE && (
```
Comment (jaixchaudhary) [inline likeminds-feed-reactnative-integration/components/LMMedia/LMCarousel/index.tsx:70]: why did you remove enum of Attachment type?

Comment (prabha-likeminds) [reply likeminds-feed-reactnative-integration/components/LMMedia/LMCarousel/index.tsx:70]: Its the same only, in UI and core they were with different names. So, now made it same everywhere.

File: likeminds-feed-reactnative-integration/components/LMPost/LMPostContent/index.tsx [hunk 3]
```diff
@@ -69,19 +71,19 @@ const LMPostContent = React.memo(({
...
-              showMoreText?.textStyle,
+              postListStyle?.postContent?.showMoreText?.textStyle,
             ])}
           >
             {showText
               ? ""
-              : showMoreText?.children
-              ? showMoreText.children
+              : postListStyle?.postContent?.showMoreText?.children
+              ? postListStyle?.postContent?.showMoreText.children
```
Comment (jaixchaudhary) [inline likeminds-feed-reactnative-integration/components/LMPost/LMPostContent/index.tsx:80]: make some const variable, don't put same check twice

File: likeminds-feed-reactnative-integration/components/LMPost/LMPostContent/index.tsx [hunk 4]
```diff
@@ -5,21 +5,20 @@ import {
...
-}: LMPostContentProps) => {
-  const MAX_LINES = visibleLines
-    ? visibleLines
+const LMPostContent = React.memo(() => {
+  const { post }: LMPostContextValues = useLMPostContext();
+  const LMFeedContextStyles = useLMFeedStyles();
+  const { postListStyle } = LMFeedContextStyles;
+
+  const MAX_LINES = postListStyle?.postContent?.visibleLines
+    ? postListStyle?.postContent?.visibleLines
```
Comment (jaixchaudhary) [inline likeminds-feed-reactnative-integration/components/LMPost/LMPostContent/index.tsx:21]: make some const variable, don't put same check twice

File: likeminds-feed-reactnative-integration/components/LMPost/LMPostFooter/index.tsx [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+              iconUrl: postListStyle?.footer?.likeIconButton?.icon?.iconUrl,
+              iconStyle: postListStyle?.footer?.likeIconButton?.icon?.iconStyle,
+              color: postListStyle?.footer?.likeIconButton?.icon?.color,
+              height: postListStyle?.footer?.likeIconButton?.icon?.height
+                ? postListStyle?.footer?.likeIconButton.icon.height
+                : 20.5,
+              width: postListStyle?.footer?.likeIconButton?.icon?.width
+                ? postListStyle?.footer?.likeIconButton.icon.width
+                : 20.5,
+              boxFit: postListStyle?.footer?.likeIconButton?.icon?.boxFit,
```
Comment (jaixchaudhary) [inline likeminds-feed-reactnative-integration/components/LMPost/LMPostFooter/index.tsx:59]: make const varibales for styles

File: likeminds-feed-reactnative-integration/context/postItemContext.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    navigation: any;
+    post: LMPostUI;
+    headerProps?: LMPostHeaderProps;
+    footerProps?: LMPostFooterProps;
+    contentProps?: LMPostContentProps;
+    mediaProps?: LMPostMediaProps;
+  }
+  
+  export interface LMPostContextValues {
+    // navigation: NativeStackNavigationProp<RootStackParamList, 'UniversalFeed'>;
```
Comment (jaixchaudhary) [inline likeminds-feed-reactnative-integration/context/postItemContext.tsx:24]: remove unnecessary comments

File: likeminds-feed-reactnative-integration/screens/createPost/index.tsx [hunk 7]
```diff
@@ -136,20 +139,41 @@ const CreatePostComponent = React.memo(() => {
...
+          placeholderText={
+            createPostStyle?.createPostTextInputStyle?.placeholderText
+              ? createPostStyle?.createPostTextInputStyle?.placeholderText
+              : CREATE_POST_PLACEHOLDER_TEXT
+          }
+          placeholderTextColor={
+            createPostStyle?.createPostTextInputStyle?.placeholderTextColor
+              ? createPostStyle?.createPostTextInputStyle?.placeholderTextColor
+              : "#0F1E3D66"
+          }
```
Comment (jaixchaudhary) [inline likeminds-feed-reactnative-integration/screens/createPost/index.tsx:151]: make const variables for styles and reuse it

File: likeminds-feed-reactnative-integration/screens/likesList/index.tsx [hunk 8]
```diff
@@ -30,16 +26,32 @@ const PostLikesListComponent = React.memo(() => {
...
-        heading="Likes"
+        {...postLikesListStyle?.screenHeader}
+        showBackArrow={
+          postLikesListStyle?.screenHeader?.showBackArrow != undefined
+            ? postLikesListStyle?.screenHeader?.showBackArrow
+            : true
+        }
+        heading={
+          postLikesListStyle?.screenHeader?.heading
+            ? postLikesListStyle?.screenHeader?.heading
```
Comment (jaixchaudhary) [inline likeminds-feed-reactnative-integration/screens/likesList/index.tsx:41]: make const variables for styles and reuse it

File: likeminds-feed-reactnative-integration/screens/postDetail/index.tsx [hunk 9]
```diff
@@ -220,7 +213,7 @@ const PostDetailComponent = React.memo(() => {
                                 repliesResponseCallback,
                                 1
                               );
-                              postDetailStyle?.commentItemStyle?.onTapReplies();
+                              postDetailStyle?.commentItemStyle?.onTapReplies && postDetailStyle?.commentItemStyle?.onTapReplies();
```
Comment (jaixchaudhary) [inline likeminds-feed-reactnative-integration/screens/postDetail/index.tsx:216]: make const variables for styles and reuse it across the file

File: likeminds-feed-reactnative-integration/components/LMPost/LMPostMedia/index.tsx [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+              <LMVideo
+                videoUrl={getUrl(VIDEO_ATTACHMENT_TYPE)}
+                {...postListStyle?.media?.video}
+              />
+            )}
+            {post?.attachments?.find(
+              (item) => item?.attachmentType === DOCUMENT_ATTACHMENT_TYPE
+            ) && (
+              <LMDocument
+                attachments={getData(DOCUMENT_ATTACHMENT_TYPE)}
```
Comment (jaixchaudhary) [inline likeminds-feed-reactnative-integration/components/LMPost/LMPostMedia/index.tsx:138]: make const variables for styles and reuse it across the file

File: likeminds-feed-reactnative-integration/components/LMPost/LMPostHeader/index.tsx [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+                postListStyle?.header?.profilePicture?.fallbackTextStyle,
+            }}
+            imageUrl={post?.user?.imageUrl}
+            onTap={postListStyle?.header?.profilePicture?.onTap}
+            size={postListStyle?.header?.profilePicture?.size}
+            fallbackTextBoxStyle={
+              postListStyle?.header?.profilePicture?.fallbackTextBoxStyle
+            }
+            profilePictureStyle={
+              postListStyle?.header?.profilePicture?.profilePictureStyle
```
Comment (jaixchaudhary) [inline likeminds-feed-reactnative-integration/components/LMPost/LMPostHeader/index.tsx:75]: make const variables for styles and reuse it across the file

File: likeminds-feed-reactnative-integration/components/LMCommentItem/index.tsx [hunk 12]
```diff
@@ -0,0 +1,435 @@
...
+              icon={{
+                assetPath: commentIsLiked
+                  ? likeIconButton?.activeIcon?.assetPath
+                    ? likeIconButton.activeIcon.assetPath
+                    : require("../../assets/images/heart_red_icon3x.png")
+                  : likeIconButton?.icon?.assetPath
+                  ? likeIconButton.icon.assetPath
+                  : require("../../assets/images/heart_icon3x.png"),
+                iconUrl: likeIconButton?.icon?.iconUrl,
+                iconStyle: likeIconButton?.icon?.iconStyle,
```
Comment (jaixchaudhary) [inline likeminds-feed-reactnative-integration/components/LMCommentItem/index.tsx:198]: make const variables for styles and reuse it across the file

**Review by jaixchaudhary — APPROVED** [review]
